engine/classes/ElggFile.php

```diff
@@ -318,6 +318,34 @@ public function tell() {
 	}
 
 	/**
+	 * Sets access and modification time of file and clears stats cache for the file
+	 * 
+	 * @param int $time  Modification time (defaults to time())
+	 * @param int $atime Access time (defaults to $time)
+	 * @return bool
+	 */
+	public function setModifiedTime($time = null, $atime = null) {
+		if (!isset($time)) {
+			$time = time();
+		}
+		if (!isset($atime)) {
+			$atime = $time;
+		}
+		if ($result = touch($this->getFilenameOnFilestore(), $time, $atime)) {
```

Comment:
Is setting particular time a really need? Seems like something that might be unreliable on some file systems.